### ISSUE TITLE
PHP Notice: Array to string conversion in /wp-content/plugins/mailgun/includes/wp-mail.php on line 157

### ISSUE THREAD

**Issue by iamphilrae** [mailgun/wordpress-plugin#10]
Please sort out the issue in v1.4 where you are not checking for an array on line 157 in wp-mail.php. This is filling up our PHP log files and essentially making them noisy and so useless.

Thanks

```
PHP Notice: Array to string conversion in /wp-content/plugins/mailgun/includes/wp-mail.php on line 157
```

**Comment by filljoyner** [comment]
bump

**Comment by lukedanielson** [comment]
Having this issue as well.  For what it's worth, i've var dumped line 157 to see what was returning:

```
var_dump($body['o:tag']);
var_dump($tags);
$body['o:tag'].=','.$tags;
```

produced:

```
array(1) {
  [0]=> string(12) "mydomain.com"
}
array(1) {
  [0]=> string(16) "mydomain-website"
}
```

So the error would be thrown on both `$body['o:tag']` and `$tags`, right?

**Comment by lukedanielson** [comment]
I am using composer.json to manage this plugin:

```
...
  "require": {
    ...
   "wpackagist-plugin/mailgun": "1.4.1",
    ...
  }
...
```

However I don't believe this repo is tagging the releases like it is in Subversion, so we can't really submit PRs.

**Comment by lukedanielson** [comment]
For what it's worth, I've edited the relevant section for an array check and it seems to be functioning fine now:

```php
    // campaign-id now refers to a list of tags which will be appended to the site tag
	if ( isset( $mailgun['campaign-id'] ) ){
		$tags = explode(",", str_replace(" ","", $mailgun['campaign-id']));
		if (empty($body['o:tag'])) {
			$body['o:tag']= $tags;
		} else if (is_array($body['o:tag'])) {
			$body['o:tag'] = array_merge($body['o:tag'], $tags);
		} else {
			$body['o:tag'].=','.$tags;
		}
	}
```

Is it okay if the `$body['o:tag']` is returned as either an array or a string?  Can this be added to the next update?

Thanks for the plugin and all the work that went into it before this!

**Comment by levymetal** [comment]
Thanks for that fix @lukedanielson!

Mailgun, any ideas when this will be merged into master? Are you still maintaining this plugin?

**Comment by andcam** [comment]
bump (@travelton  @BFTrick  @matt-h has this been abandoned?)

**Comment by pirogoeth** [comment]
I have fixed this issue and the 1.5 release is going out today. :)
